### DISCUSSION TITLE
Combine .drawio XML with SVG in single file

### DISCUSSION THREAD

**Discussion by brandonkirsch** [jgraph/drawio#4226] [Ideas]
I see that this repo is not open to contributions, so instead I will throw out a wild idea that would make drawio files more useful to developers.

What if drawio supported an export format that appeared like an SVG file, but also contained the actual diagram content as well embedded in an <foreignElement> SVG tag.

When included on a webpage, the SVG image would render the diagram.  But then when opened with the drawio app, the original diagram contents would be available for editing and making future changes.

I think that would be so slick to have a file format that acted as an image but also contained the structured data needed to open & edit.  It would work very nicely in Git and could be directly included and rendered in HTML documentation.

**Comment by davidjgraph** [comment]
Maybe it would have been worth trying an export to SVG yourself before posting, then you'd have seen we already do that...

<img width="457" alt="Capture d’écran 2024-02-29 à 21 20 03" src="https://github.com/jgraph/drawio/assets/1761087/e88b5849-5aec-4d21-8489-b84d266dcb20">

**Comment by brandonkirsch** [comment]
That's absolutely fantastic!  This works extremely well with Gitlab and **greatly** increases the practicality and usefulness of drawio.

![image](https://github.com/jgraph/drawio/assets/449783/27a988e5-c197-4278-9513-29433f827c8f)

Great work!!
